input-explainer.md

```diff
@@ -177,13 +177,27 @@ Most applications will want to visually represent the input sources somehow. The
   * `'tracked-pointer'`: If the `gripSpace` is not `null` an application-appropriate controller model should be drawn using that transform. If appropriate for the experience, the a visualization of the target ray and a cursor as described in the `'gaze'` should also be drawn.
   * `'screen'`: In all cases the point of origin of the target ray is obvious and no visualization is needed.
 
+Determining what to render is important as well. In some cases it's appropriate for the application to render a contextually appropriate model (such as a racket in a tennis game), in which case the physical appearance of the input source doesn't matter much. Many other times, though, it's desirable for the application to display a device that matches what the user is holding, especially when relaying instructions about it's use. Finally there are times when it's best to not render anything at all, such as when the XR device uses a transparent display and the user can see their hands and/or any tracked devices without app assistance.
+
+The `XRInputSource`'s `renderId` is used to determine what should be rendered in the cases when an application specific model isn't being used. 
+
+ - A `renderId` of `''` (empty string) indicates the input source should not be rendered. Cursors and pointing rays should still be rendered as described above if needed by the application. Input sources with a `targetRayMode` of `'gaze'` or `'screen'` MUST have a `renderId` of `''`. `'tracked-pointer'` input sources should typically only have a `renderId` of `''` when using displays that allow the user to see their environment and the physical input source.
+ - The `renderId` may be `'unknown'` if the type of input source cannot be reliably identified or the UA determines that the input source type must be masked for any reason.
+ - Otherwise the `renderId` should be a lower-case string that describes the physical input source. It should not include and indication of the handedness of the input source, as that should be implied by the `handedness` attribute.
+   - If the input source is a tracked hand without any held controller, the `renderId` MUST be `hand`
+   - For most controllers this should ideally be of the format `<vendor>-<product-id>`. For example: `oculus-touch`. UAs should make an effort to align on the strings that are returned for any given device.
```

Comment:
> UAs should make an effort to align on the strings that are returned for any given device.

We should explicitly dig in on this at the F2F to work out some clever approach for going about this even without lots of cross-UA coordination before the first UA ships support for a device.

Reply:
https://www.w3.org/wiki/Repositories appears to be the latest thinking on such things and seems like a good fit.

I think this is a large enough topic that we should break the renderId part out of this PR so we can focus on controller state ergonomics and the logistics around this ID separately.

Reply:
I'd caution against taking the thinking in that wiki as a current, direct solution.  It was a work item of the Advisory Board (while I was on it), hasn't been completed (or worked on in a while?) and it's not an adopted part of the W3C Process (translation: this hasn't been tested by any other repository/schema-like thing in W3C-space that I know of).